### ISSUE TITLE
Facebook authentication from iPhone not setting timezone/permissions correctly

### ISSUE THREAD

**Issue by loarie** [inaturalist/INaturalistIOS#225]
Unlike Facebook authentication from the Web https://github.com/inaturalist/inaturalist/issues/314 (which does grab the timezone and license content)
authentication from the iPhone does not:
<img width="746" alt="screen shot 2015-11-04 at 10 10 00 pm" src="https://cloud.githubusercontent.com/assets/598026/10961225/5ea29496-8341-11e5-86bf-46ce2083c1f5.png">
<img width="746" alt="screen shot 2015-11-04 at 10 10 38 pm" src="https://cloud.githubusercontent.com/assets/598026/10961223/5b6919e4-8341-11e5-893f-c43bab03df5a.png">
it should default to CC licenses for photos and obs and set the timezone

**Comment by carrieseltzer** [comment]
Thanks, Scott! I'm pretty sure this is happening for Android too (see example: http://www.inaturalist.org/observations/2346706) but I can't test it. And is this NOT happening for G+ authentication?

**Comment by loarie** [comment]
It just occurred to me that this isn't a bug (switched label to improvement). Since we don't show mobile user the 'Yes license my content' checkbox in the FB and Google signup flows, we're just being conservative here and assuming they don't want to license their content.

If we added the ability to opt out to the app settings, I think it would be safe to rather opt these users into the default licenses. 

So I propose we:
1. apply default (ie 'Yes license my content' ) license to content for signups through FB/Google

2. Add License Picker to Settings screen which sets the users license preferences (does not update past content)
Title: 'Media License'
Description: 'Choose a license to apply to the photos I post (visit inaturalist.org to update past photos with a different choice)'
Choices: CC-BY, CC-BY-NC, CC-BY,SA, CC-BY-ND, CC-BY-NC-SA, CC-BY-NC-ND, no license

Adding a caveat though that this might not be a priority. Looks like not that people sign up through FB/Google
```
#users in last 6 months
User.where("created_at > ?", 6.months.ago).count
=> 52893 

#number of FB, Google accounts from users in the last 6 months, ie ~10% of new users used FB to sign up
ProviderAuthorization.joins('LEFT JOIN users on user_id = users.id').where("users.created_at > ?", 6.months.ago).group('provider_name').count
=> {"soundcloud"=>59, "google_oauth2"=>3972, "facebook"=>5907, "twitter"=>181, "flickr"=>243} 

#no way of knowing which ones signed up through mobile, but only 19 people who signed up in the last 6 months using FB have the obs_license set to nil
Preference.joins('LEFT JOIN provider_authorizations on owner_id = provider_authorizations.user_id').joins('LEFT JOIN users on owner_id = users.id').where("value IS NULL AND preferences.name = ? AND users.created_at > ? AND owner_type = ?", "observation_license", 6.months.ago, "User").group('provider_authorizations.provider_name').count
=> {nil=>70, "soundcloud"=>2, "google_oauth2"=>7, "facebook"=>19, "flickr"=>6, "twitter"=>2} 
```

**Comment by carrieseltzer** [comment]
Makes sense. I'm surprised though that the number w/o licenses is so low... I practically feel like I could find you 19 users myself! I feel like I come across about 1 new user per day with ARR on their observations. 




I think the proposed solution to add a setting in the app is great. 



—

**Comment by alexshepard** [comment]
verify that signup using Facebook and/or Google use default permissions for sharing.

**Comment by kueda** [comment]
I think 1 should be happening (Alex will verify), I think 2 is overkill for the apps, since I think a tiny fraction of people are interested in that, and those that are probably use a combo of SLR / website. Is that really that important, and if so, for who?
